pkg/sql/lease.go

```diff
@@ -98,6 +98,15 @@ func (s *tableVersionState) hasExpired(timestamp hlc.Timestamp) bool {
 	return !timestamp.Less(s.expiration)
 }
 
+// leaseHasEnoughLifeLeft checks whether the table descriptor lease has
+// enough life: that it expires further then LeaseRenewalTimeout
+// duration from now.
+func (s *tableVersionState) leaseHasEnoughLifeLeft() bool {
+	// Get the current time and move it forward by the lease renewal timeout.
+	t := hlc.Timestamp{WallTime: hlc.UnixNano() + int64(LeaseRenewalTimeout)}
+	return t.Less(s.expiration)
```

Comment:
I'd return !t.Less(s.expiration) here and call this method shouldPreemptivelyRenewLease(). The current name is easy to confuse with hasExpired()

Reply:
I also think you can use the clock in tableVersionState instead of calling hlc.UnixNano()